app/src/main/java/com/example/plantsapp/presentation/ui/notification/TaskNotificationManager.kt

```diff
@@ -22,12 +28,24 @@ class TaskNotificationManager @Inject constructor(
     @ApplicationContext private val context: Context
 ) {
     private var notificationId = 0
-    private val pendingIntent = createPendingIntent()
+    private val clickPendingIntent by lazy {
+        PendingIntent.getActivity(
+            context,
+            NOTIFICATION_REQUEST_CODE,
+            Intent(context, MainActivity::class.java),
+            PendingIntent.FLAG_IMMUTABLE
+        )
+    }
 
     init {
         createChannel(context)
     }
 
+    fun cancelNotification(notificationId: Int) {
+        (context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager)
```

Comment:
create private getter for this NotificationManager (it is used in 2 places in this class)
private val notificationManager get() = ....